doc/overall_design.md

```diff
@@ -0,0 +1,58 @@
+# ElasticDL Overall Design
+
+## Component Architecture
+
+![component_architecture](/doc/figures/component_architecture.png)
+
+ElasticDL uses the master-worker architecture. The master node plays the master role in two aspects.
+
+1. It's the master of the cluster. It manages the lifecycle of all the worker pods, starts the worker pod, listens to the pod event and relaunches the terminated worker pod if necessary.
+2. It's the master of the model training process.
+   * Partition the training/evaluation data into mutiple shards.
+   * Generate the training/evaluation tasks from the data shards.
+   * Aggregate the gradients reported from the workers.
+   * Update the model variables and save the checkpoint if necessary.
+
+ElasticDL client is simple, just like a CLI command. User types ElasticDL command in the terminal to start the training/evaluation/prediction job. The client parses the parameters, builds the image which packages the ElasticDL framework and the model code, pushes the image into the hub, and then applies the CRD of the master node to the kubernetes ApiServer. After the master pod is created and started, it will then create other components and drive the process of the entire job.
```

Comment:
There is no CRD involved in ElasticDL since ElasticDL is kubernetes-native.

Reply:
Good idea, rename CRD to "send request to create the pod"